types/time.go

```diff
@@ -2349,12 +2341,21 @@ func parseTwoNumeric(input string) (int, string) {
 }
 
 func dayOfMonthNumeric(t *MysqlTime, input string, ctx map[string]int) (string, bool) {
-	v, remain := parseTwoNumeric(input) // 0..31
-	if len(remain) == len(input) || v > 31 {
+	result := digitRegex.FindString(input) // 0..31
+	length := len(result)
+
+	if length > 2 {
```

Comment:
is it better to add some comments here why do you only take the suffix after index 2?